### PR TITLE
Readme Updates

### LINKED ISSUE #111: [BUG] Cargo build fails in a vscode devcontainer running on MacOS (x64) with Docker VirtuoFS
Running **"cargo build -p chariott"** fails in a vscode devcontainer running on MacOS (x64) with Docker VirtuoFS.

## Expected Behavior
Cargo build should complete with success

## Current Behavior
Build fails printing various inconsistent error messages typically complaining not being able to locate artifacts in the target folder:

<reducted>
...
  = note: /usr/bin/ld: cannot find /workspaces/chariott/target/debug/build/chariott-proto-081e3239d3e38e03/build_script_build-081e3239d3e38e03.376i4vsos7gxedbo.rcgu.o: No such file or directory
          /usr/bin/ld: cannot find /workspaces/chariott/target/debug/build/chariott-proto-081e3239d3e38e03/build_script_build-081e3239d3e38e03.ybchmlo3fea63ob.rcgu.o: No such file or directory
          collect2: error: ld returned 1 exit status
          
error: could not compile `chariott-proto` due to previous error

## Steps to Reproduce

1. Clone charriot repository from github
2. Launch vscode remote devcontainer
3. cargo build -p chariott

Build fails

## Context (Environment)

 * Running with vscode ( 1.76.2) 
 * devcontainer (0.2888)
 * Docker Desktop v4.16.2  (with VirtuoFS enabled)
 * MacBook Pro (i9 Intel)  macos Ventura 13.2.1.

## Additional Information

This is not a bug report but a request to extend the README documentation for mac users. After quite a bit trial and error and searching similar problems with rust in devcontainers, the following  [Stack Overflow solution](https://stackoverflow.com/questions/72448053/rust-incremental-build-not-working-in-vscode-devcontainer) provided the correct diagnosis and a workaround solution.

I am not sure this needs to be a fix in this repository or chariott but I decided to log it as an issue that should probably be a part of the README file to help others using a Macbook to build charriot.

Docker Desktop provides a fast VirtuoFS implementation on a mac but rust build process breaks in devcontainers if VirtuoFS is enabled.  The solution is not disabling the VirtuoFS, because this significantly slows down the I/O operations in the containers.  

The workaround is to create a target folder  outside the chariott workspace and set the environment variable to the new target folder. The following sequence works in the dev container:

```sh
vscode ➜ /workspaces/chariott (main) $ sudo mkdir ../target
vscode ➜ /workspaces/chariott (main) $ sudo chown vscode:vscode ../target
vscode ➜ /workspaces/chariott (main) $ CARGO_TARGET_DIR=../target cargo build -p cargo
vscode ➜ /workspaces/chariott (main) $ CARGO_TARGET_DIR=../target cargo run -p chariott
```

### PR DESCRIPTION
**Pull request by ladatz**
Closes #111 

## Motivation and Context
Some README dependency and link updates

## Description

- Adds information from issue 111 for how to build with MacOS and VirtuoFS
- Adds link to service discovery doc
- Adds/ clarifies some dependency instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +97,25 @@ cargo build --workspace
 cargo test --workspace
 ```
 
+#### Build on MacOS with Docker VirtuoFS
+
+As reported in [this issue](https://github.com/eclipse-chariott/chariott/issues/111), some extra
+steps are needed to build if you are using MacOS with Docker VirtuoFS.
+
+Docker Desktop provides a fast VirtuoFS implementation on a Mac but rust build process breaks in
```
Comment (devkelley) [inline README.md:105]:
```suggestion
Docker Desktop provides a fast VirtuoFS implementation on a Mac but the rust build process breaks in
```

File: README.md [hunk 2]
```diff
@@ -105,6 +97,25 @@ cargo build --workspace
...
+devcontainers if VirtuoFS is enabled. The solution is not disabling the VirtuoFS, because this
+significantly slows down the I/O operations in the containers.
+
+The workaround is to create a target folder outside the chariott workspace and set the environment
+variable to the new target folder. The following sequence works in the devcontainer:
+
+```shell
+vscode ➜ /workspaces/chariott (main) $ sudo mkdir ../target
+vscode ➜ /workspaces/chariott (main) $ sudo chown vscode:vscode ../target
+vscode ➜ /workspaces/chariott (main) $ CARGO_TARGET_DIR=../target cargo build -p cargo
```
Comment (ladatz) [inline README.md:115]: This should say cargo build -p chariott

Comment (ashbeitz) [inline README.md:115]: I think that these 2 lines should be:

export CARGO_TARGET_DIR=../target
cargo build -p cargo
cargo run -p chariott

Comment (ashbeitz) [reply README.md:115]: You may be able to drop the export when it runs in vscode.

**Review by ashbeitz — COMMENTED** [review]
Please take a look at my comments.

File: README.md (hunk 2 above)
Comment (ladatz) [reply README.md:115]: As discussed, this is a valid way of specifying the cargo target directory (https://stackoverflow.com/questions/50364390/how-can-i-specify-a-custom-cargo-output-directory). Given that this was a fix suggested by another contributor and we've validated that these commands work, I will leave it as-is.

**Review by ashbeitz — APPROVED** [review]
Approved

**Pull request merged**